test/headersReceived.test.js

```diff
@@ -1,281 +1,229 @@
+            test("header is invalid value", () => {
+                const header = {name: CHL_BYPASS_SUPPORT, value: "0"};
+                found = isBypassHeader(header);
+                expect(found).toBeFalsy();
+            });
+            test("header is invalid name", () => {
+                const header = {name: "Different-header-name", value: `${configId}`};
+                found = isBypassHeader(header);
+                expect(found).toBeFalsy();
+            });
+            test("config is reset if ID changes", () => {
+                const oldConfigId = configId + 1;
+                workflow.__with__({CONFIG_ID: oldConfigId})(() => {
+                    setMock(bypassTokensCount(oldConfigId), 10);
+                    const header = {name: CHL_BYPASS_SUPPORT, value: `${configId}`};
+                    const oldCount = getMock(bypassTokensCount(oldConfigId));
+                    found = isBypassHeader(header);
+                    expect(found).toBeTruthy();
+                    expect(oldCount === getMock(bypassTokensCount(configId))).toBeFalsy();
+                });
+            });
+            test("config is not reset if ID does not change", () => {
+                const oldConfigId = configId;
+                workflow.__with__({CONFIG_ID: oldConfigId})(() => {
+                    setMock(bypassTokensCount(oldConfigId), 10);
+                    const header = {name: CHL_BYPASS_SUPPORT, value: `${configId}`};
+                    const oldCount = getMock(bypassTokensCount(oldConfigId));
+                    found = isBypassHeader(header);
+                    expect(found).toBeTruthy();
+                    expect(oldCount === getMock(bypassTokensCount(configId))).toBeTruthy();
+                });
+            });
         });
-
-        describe("signing enabled", () => {
+        describe("check redemption attempt conditions", () => {
+            let url;
+            let details;
+            let header;
             beforeEach(() => {
-                workflow.__set__("DO_SIGN", true);
-                workflow.__set__("readySign", false);
+                header = {name: CHL_BYPASS_SUPPORT, value: configId};
+                details = {
+                    statusCode: spendStatusCode()[0],
+                    responseHeaders: [header],
+                };
+                url = new URL("http://www.example.com");
             });
 
-            test("no tokens", () => {
+            test("check that favicon urls are ignored", () => {
+                url = new URL("https://example.com/favicon.ico");
+                expect(isFaviconUrl(url.href)).toBeTruthy();
                 const fired = processHeaders(details, url);
                 expect(fired).toBeFalsy();
-                const readySign = workflow.__get__("readySign");
-                expect(readySign).toBeTruthy();
-                expect(updateIconMock).toBeCalledWith("!");
+                expect(updateIconMock).toBeCalledTimes(0);
```

Comment:
This seems to have changed value (1 -> 0)? I find this strange, as it implies there is a functional difference in the code somewhere. Maybe it is something to do with `setConfig()`

Reply:
I think you're right, `setConfig` likely caused a redraw of the token number, any preferences to addressing this, the functionality makes sense, I'll see about adding it back.

Reply:
I'm just eager to ensure that this change (given the size of it) does not result in a change to any of the functionality, and I think that this may indicate that the workflow has been altered somewhere?

Reply:
Actually, I think this is fine to ignore, I think it's just because setConfig is no longer called before each of the tests are ran.

Having said that, I noticed that some of the tests have had the `updateIcon` check removed (e.g. `"config is reset if ID changes"`), is there a reason for that?

Reply:
I'm good to ignore, let me look into these `updateIcon` removals, might be an error

Reply:
Not sure, why those got removed, put them back